### ISSUE TITLE
Enable Hot-reload for CLI

### ISSUE THREAD

**Issue by samad-yar-khan** [middlewarehq/middleware#423]
## Why do we need this ?

- The current dev experience for the main app is made super convenient using `./dev.sh` which triggers the CLI to manage the Docker containers
- The dev experience for the CLI itself is not convenient as simple changes to the CLI require a reload.

## Acceptance Criteria

- [ ] Enable Hot reloading on the CLI
- [ ] Make sure the hot reload to the CLI itself does not take too long and does not build the containers again from scratch

**Comment by VipinDevelops** [comment]
Hey @samad-yar-khan  Can you explain this issue in more detail

Currently we build and start the CLI its like starting point of the development server that handles everything and  we have `dev` script in the CLI if someone want to run that in watch mode with hot reload.

so Can you please explain are we trying to achieve here ?

**Comment by jayantbh** [comment]
@VipinDevelops this issue basically meant that when you're working on a change in the CLI itself, instead of shutting down and restarting all changes, it should have a live-reload functionality similar to how you'd have experienced in the usual UI dev.

**Comment by VipinDevelops** [comment]
Hey @jayantbh Thanks for explaining, but we already have a dev scripting in the CLI that can run it with hot reload.

**Comment by jayantbh** [comment]
@VipinDevelops does it actually work like live-reload for you? Because it seems to basically trigger a whole-app-reload for the CLI for me on changes.

**Comment by VipinDevelops** [comment]
So @jayantbh correct me if I am wrong we want to make a feature that lets use inject updated source code into a running app without a restart

**Comment by jayantbh** [comment]
Correct. The running app here is the CLI app. Not the frontend or backend code.

**Comment by VipinDevelops** [comment]
Got it and that code with run other services thanks for explaining I will try to explore potential solution for this.
